packages/beacon-node/src/chain/chain.ts

```diff
@@ -177,6 +179,8 @@ export class BeaconChain implements IBeaconChain {
   protected readonly blockProcessor: BlockProcessor;
   protected readonly db: IBeaconDb;
   private readonly archiver: Archiver;
+  // this is only available if nHistoricalStates is enabled
```

Comment:
is there any reason we still wanna keep the in-memory state management?

Reply:
cc @twoeths I am still curious about this

Reply:
ah good question, I don't see any benefit to use it from our side
if any consumers want to use proof api, it's not always available with our persistent checkpoint state cache. But we haven't received any complaints since we switch to this cache
so I guess we can remove it at some points to simplify our source code